src/common/runnable/run-many-for.ts

```diff
@@ -3,46 +3,81 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
```

Comment:
This commit for a new feature of `Runnable` is unnecessarily difficult to read without accompanying unit tests.